python/triton/language/core.py

```diff
@@ -670,9 +671,10 @@ def is_ptr(self):
     def is_const(self):
         return self.const
 
-    def __eq__(self, other: pointer_type) -> bool:
-        if not isinstance(other, pointer_type):
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, (pointer_type, constexpr)):
             return False
+        other = _unwrap_if_constexpr(other)
```

Comment:
You're assuming that constexpr will always contain a `pointer_type`
```suggestion
        other = _unwrap_if_constexpr(other)
        if not isinstance(other, pointer_type):
            return False
```